src/main/java/maria/Parser.java

```diff
@@ -0,0 +1,115 @@
+import maria.command.CommandFindTask;
+import maria.command.CommandHelp;
+import maria.command.CommandListAllTasks;
+import maria.command.CommandMarkTask;
+import maria.command.CommandRemoveTask;
+import maria.command.CommandUnknown;
+import maria.command.CommandUnmarkTask;
+
+/**
+ * Parses commands.
+ */
+public class Parser {
+
+    public static Command parse(String commandStr) {
+
+        try {
+
+            String[] commandStrArr = commandStr.split(" ");
+            String commandMain = commandStrArr[0];
+
+            switch (commandMain) {
+            case "bye":
+                return new CommandExit();
+            case "todo": {
+                String name = commandStrArr[1];
+                boolean done = Boolean.parseBoolean(commandStrArr[2]);
+                return new CommandAddTodo(name, done);
+            }
+            case "deadline": {
+                String name = commandStrArr[1];
+                boolean done = Boolean.parseBoolean(commandStrArr[2]);
+                LocalDate deadline = LocalDate.parse(commandStrArr[3]);
+                return new CommandAddDeadline(name, done, deadline);
+            }
+            case "event": {
+                String name = commandStrArr[1];
+                boolean done = Boolean.parseBoolean(commandStrArr[2]);
+                LocalDate start = LocalDate.parse(commandStrArr[3]);
+                LocalDate end = LocalDate.parse(commandStrArr[4]);
+                return new CommandAddEvent(name, done, start, end);
+            }
+            case "list":
+                return new CommandListAllTasks();
+            case "mark": {
+                int index = Integer.parseInt(commandStrArr[1]);
+                return new CommandMarkTask(index);
+            }
+            case "unmark": {
+                int index = Integer.parseInt(commandStrArr[1]);
+                return new CommandUnmarkTask(index);
+            }
+            case "delete": {
+                int index = Integer.parseInt(commandStrArr[1]);
+                return new CommandRemoveTask(index);
+            }
+            case "find": {
+                String searchString = commandStrArr[1];
+                return new CommandFindTask(searchString);
+            }
+            case "help": {
+                return new CommandHelp();
+            }
+            default:
+                return new CommandUnknown();
+            }
+
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+            return new CommandUnknown();
+        }
+
+    }
+
+    /**
+     * Parses the string given by the storage file.
+     * @param storageStr The storage string read from file
+     * @return A Command object that can be executed
+     */
+    public static Command parseStorage(String storageStr) {
+
+        try {
+            String[] tokens = storageStr.split("\\|\\|\\|");
+            String name = tokens[0];
+            boolean done = Boolean.parseBoolean(tokens[1]);
```

Comment:
Perhaps you may want to name boolean variables such that there is a prefix in front to make them sound like booleans? So something like: 
```suggestion
            boolean isDone = Boolean.parseBoolean(tokens[1]);
```
There are a few of these in other classes as well, so you may want to consider them too